scripts/release_scripts/repo_specific_changes_fetcher_test.py

```diff
@@ -101,16 +105,22 @@ def mock_run_cmd(unused_cmd: str) -> str:
     def test_get_changes(self) -> None:
         def mock_get_changed_schema_version_constant_names(
             unused_release_tag_to_diff_against: str
-        ) -> List[str]:
-            return ['version_change']
+        ) -> Optional[List[str]]:
+            if self.call_counter:
```

Comment:
Your mock function's return value should not be dependent on self.call_counter ... that just seems odd. At most the mock should just return some hardcoded value, unless there is some clearer motivation for having the first calls return None (which the reader would obviously grok).

Reply:
Returning `None` ensures that we are handling it properly -- thus covering all possible branches,